internal/provider/resource_packer_run_task.go

```diff
@@ -0,0 +1,168 @@
+		Timeouts: &schema.ResourceTimeout{
+			Create:  &defaultPackerTimeout,
+			Default: &defaultPackerTimeout,
+			Update:  &defaultPackerTimeout,
+			Delete:  &defaultPackerTimeout,
+		},
+		CustomizeDiff: resourcePackerRunTaskCustomizeDiff,
+		Schema: map[string]*schema.Schema{
+			// Optional inputs
+			"regenerate_hmac": {
+				Description: "If true, the HMAC Key (`hmac_key`) will be regenerated during `terraform apply` and the resource will always cause a non-empty plan. Changing `regenerate_hmac` to false (or removing it from the config) should not result in a plan.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+			},
+			"project_id": {
+				Description: `
+The ID of the HCP project where the HCP Packer Registry is located. 
+If not specified, the project specified in the HCP Provider config block will be used, if configured.
+If a project is not configured in the HCP Provider config block, the oldest project in the organization will be used.`,
+				Type:         schema.TypeString,
+				Computed:     true,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.IsUUID,
+			},
+			// Computed Values
+			"organization_id": {
+				Description: "The ID of the HCP organization where this channel is located. Always the same as the associated channel.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"endpoint_url": {
+				Description: "A unique HCP Packer URL, specific to your HCP organization and HCP Packer registry. The Terraform Cloud run task will send a payload to this URL for image validation.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"hmac_key": {
+				Description: "A secret key that lets HCP Packer verify the run task request.",
+				Sensitive:   true,
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func resourcePackerRunTaskCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*clients.Client)
+	loc, err := getAndUpdateLocationResourceData(d, client)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	resp, err := clients.GetRunTask(ctx, client, loc)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.SetId(loc.ProjectID)
+	if err := d.Set("endpoint_url", resp.APIURL); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("hmac_key", resp.HmacKey); err != nil {
+		return diag.FromErr(err)
+	}
+
+	regenerateHmac, ok := d.GetOk("regenerate_hmac")
+	if ok && regenerateHmac.(bool) {
+		resp, err := clients.RegenerateHMAC(ctx, client, loc)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		if err := d.Set("hmac_key", resp.HmacKey); err != nil {
+			return diag.FromErr(err)
+		}
+		// Always set `regenerate_hmac` to false so that it can be set to false by
+		// the user without generating a diff.
+		if err := d.Set("regenerate_hmac", false); err != nil {
+			return diag.FromErr(err)
```

Comment:
What happens if you don't do this?

Reply:
Since `regenerate_hmac` is intended to be a one-time operation, this automatically toggles it back to false in the state.

- With this code
  - A plan will always be generated when `regenerate_hmac = true`. 
    - Plans show that `hmac_key` will regenerate
    - Plans with `regenerate_hmac = true` will always show that the state contains `regenerate_hmac = false`, even if applied multiple times in a row
  - A plan will never be generated when `regenerate_hmac = false`. 
- Without this code
  - A plan will always be generated when `regenerate_hmac = true`. 
    - Plans show that `hmac_key` will regenerate
  - A plan _may_ be generated when `regenerate_hmac = false` _if_ the config contained `regenerate_hmac = true` during the last apply

TL;DR: This code removes the (effectively) no-op plan when applying with `regenerate_hmac = false` after previously applying with `regenerate_hmac = true`

Reply:
Nice! Thanks for explaining!